common/setups/returnn_pytorch/serialization.py

```diff
@@ -1,18 +1,22 @@
 from __future__ import annotations
-from typing import Any, Dict, List, Optional, Set, Union
+
 import os
 import pathlib
 import shutil
 import string
 import textwrap
+from dataclasses import fields
+from inspect import isfunction
+from typing import Any, Dict, List, Optional, Set, Tuple, Union
 
+import torch
+from i6_core.util import instanciate_delayed
+from i6_models.config import ModelConfiguration, ModuleFactoryV1
```

Comment:
~~I would move this import locally into the build function, otherwise this will cause issues with existing setups.~~ Ah this does not work because it is needed for the typing... hmm maybe this works?
```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from i6_models.config import ModelConfiguration, ModuleFactoryV1
```

And then import it again within the function.

Reply:
No need to import again within the function. But what needs to be done is converting the type-annotations to strings, this way the concrete type is not required here.

Reply:
> No need to import again within the function
But if you do not do that, it will not exist, so then the `isinstance` checks will fail.

Reply:
Moved import inside the function and annotated `cfg: Any` instead paired with an `assert isinstance(cfg, ModelConfiguration)` inside the function.